File: frontend/src/app/shared/models/review.ts
export interface Review {
  id: number;
  reviewerId: number;
  reviewerUsername: string;
  reviewerProfileImageUrl?: string | null;
  reviewedUserId: number;
  rating: number;
  comment: string;
  createdAt: string;
  updatedAt: string;
}

export interface PagedReviews {
  total: number;
  items: Review[];
}

export type ReviewSortOrder = 'Newest' | 'HighestRating' | 'LowestRating';

File: backend/Snagged.backend/Snagged.Domain/Entities/Review.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Snagged.Domain.Entities
{
    public class Review
    {
        public int Id { get; set; }

        public int ReviewerId { get; set; }
        public User? Reviewer { get; set; }

        public int ReviewedUserId { get; set; }
        public User? ReviewedUser { get; set; }

        public int Rating { get; set; }
        public string Comment { get; set; } = string.Empty;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;


    }
}
